tests/behat/behat-features/src/JsonApiContext.php

```diff
@@ -0,0 +1,222 @@
+      ->setHeader("Accept-Charset", "utf-8");
+
+    $request = $requestBuilder->getRequest();
+
+    $response = $this->jsonApiClient->sendRequest($request);
+
+    if (!$response->isSuccessful() || !$response->hasDocument()) {
+      throw new ExpectationException('Request to fetch an article was not successful or has no results', $this->getSession());
+    }
+
+    $hydrator = new ClassHydrator();
+    $article = $hydrator->hydrate($response->document());
+
+    $fields_list = explode(',', $fields);
+    foreach ($fields_list as $field) {
+      if (empty($article->{$field})) {
+        throw new ExpectationException("A field was not part of the response: $field", $this->getSession());
+      }
+    }
+
+  }
+
+  /**
+   * @Then I create an article with the title :title
+   */
+  public function iCreateAnArticleWithTheTitle($title) {
+    $oauth_access_token = $this->getOauthAccessToken();
+    // Instantiate an empty PSR-7 request.
+    $request = new Request("", "");
+
+    $requestBuilder = new JsonApiRequestBuilder($request);
+
+    $requestBuilder
+      ->setProtocolVersion("1.1")
+      ->setMethod("POST")
+      ->setUri("{$this->contentpoolBaseUrl}/jsonapi/node/article")
+      ->setHeader("Accept-Charset", "utf-8")
+      ->setHeader("Authorization", "Bearer $oauth_access_token");
+
+    $requestBuilder->setJsonApiBody('
+    {
+      "data": {
+        "type": "node--article",
+        "attributes": {
+          "title": "' . $title . '",
+          "field_seo_title": "' . $title . '"
+        },
+        "relationships": {
+          "field_channel": {
+            "data": {
+              "type": "taxonomy_term--channel",
+              "id": "1bc7757f-ff52-4dde-ab24-68c1cd7362b8"
+            }
+          }
+        }
+      }
+    }
+    ');
+
+    $request = $requestBuilder->getRequest();
+
+    $response = $this->jsonApiClient->sendRequest($request);
+
+    if ($response && !$response->isSuccessful()) {
+      $error = $response->document()->error(0);
+      throw new ExpectationException(sprintf('Request to create an article was not successful or has no results: %s: %s', $error->title(), $error->detail()), $this->getSession());
+    }
+
+  }
+
+  /**
+   * Retrieves an oauth2 access token.
+   *
+   * @return string
+   *   Access Token.
+   *
+   * @throws \Behat\Mink\Exception\ExpectationException
+   */
+  private function getOauthAccessToken() {
+
+    $config = [
+      "client_id" => $this->consumerId,
+      "client_secret" => "behat123",
+      "username" => "dru_admin",
+      "password" => "changeme",
+      "grant_type" => "password",
+    ];
+
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, "{$this->contentpoolBaseUrl}/oauth/token");
+    curl_setopt($ch, CURLOPT_POST, TRUE);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+    /* curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, FALSE); */
+    curl_setopt($ch, CURLOPT_POSTFIELDS, $config);
+    $data = curl_exec($ch);
```

Comment:
you could have used guzzle, but nevermind

Reply:
tried to use guzzle for oauth2 in the test, but had problems making the yang jsonapi client work with it